web-common/src/features/dashboards/state-managers/selectors/dimension-filters.ts

```diff
@@ -99,27 +99,36 @@ export function getDimensionFilterItems(
   dashData: AtLeast<DashboardDataSources, "dashboard">,
 ) {
   return (dimensionIdMap: Map<string, MetricsViewSpecDimensionV2>) => {
-    if (!dashData.dashboard.whereFilter) return [];
-
-    const filteredDimensions: DimensionFilterItem[] = [];
-    const addedDimension = new Set<string>();
-    forEachIdentifier(dashData.dashboard.whereFilter, (e, ident) => {
-      if (addedDimension.has(ident) || !dimensionIdMap.has(ident)) return;
-      const dim = dimensionIdMap.get(ident);
-      if (!dim) {
-        return;
-      }
-      addedDimension.add(ident);
-      filteredDimensions.push({
-        name: ident,
-        label: getDimensionDisplayName(dim),
-        selectedValues: getValuesInExpression(e),
-      });
+    return getDimensionFilterItemsForFilter(
+      dimensionIdMap,
+      dashData.dashboard.whereFilter,
+    );
+  };
+}
+
+export function getDimensionFilterItemsForFilter(
```

Comment:
This function name is a little confusing. Maybe  just `getDimensionFilters()` would be more readable?